### PR TITLE
Running metrics

### PR DESCRIPTION
**Pull request by wiseodd**
Support running metrics in cross validation of prior precision. Rationale: gathering data then computing metric is very expensive if `n_data` is very large.  

There are some failed tests for LLLaplace, but I think they are unrelated.

### CODE REVIEW

File: laplace/utils/metrics.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        targets: torch.Tensor
+            integer tensor of shape (...)
+        """
+        probs = probs.view(-1, probs.shape[-1])
+        targets = targets.view(-1)
+
+        self.nll_sum += F.nll_loss(
+            probs.log(), targets, ignore_index=self.ignore_index, reduction='sum'
+        ).item()
+        self.n_valid_labels += (targets != self.ignore_index).sum().item()
```
Comment (runame) [inline laplace/utils/metrics.py:37]: Same here.

File: laplace/utils/metrics.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+    def compute(self):
+        return self.nll_sum / self.n_valid_labels
```
Comment (runame) [inline laplace/utils/metrics.py:40]: This would then be `self.nll_sum.item() / self.n_valid_labels.item()`.

File: laplace/utils/metrics.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    NLL metrics that
+
+    Parameters
+    ----------
+    ignore_index: int, default = -100
+        which class label to ignore when computing the NLL loss
+    """
+    def __init__(self, ignore_index=-100):
+        super().__init__()
+        self.add_state('nll_sum', default=torch.tensor(0.), dist_reduce_fx='sum')
```
Comment (runame) [inline laplace/utils/metrics.py:17]: If we go with my suggestion, we should initialise these tensors on the used device (can be an argument to `__init__`).

**Review by runame — APPROVED** [review]
Besides my minor comment, LGTM! Can you check how many tests failed? Only 7 should fail and they all should be related to ` LowRankLaplace` (since this hasn't been implemented for the new ASDL version yet).

**Comment by wiseodd** [comment]
@runame please check again!

**Comment by runame** [comment]
LGTM, have you checked the tests?

**Comment by wiseodd** [comment]
Updated the test! @runame
